addons/hw_drivers/driver.py

```diff
@@ -14,11 +15,12 @@ def __new__(cls, clsname, bases, attrs):
             newclass.priority += 1
         else:
             newclass.priority = 0
-        drivers.append(newclass)
+        if clsname != 'Driver':  # Avoid registering the abstract Driver class itself
```

Comment:
Metaclass instantiated ?

Reply:
yes!

I think because we override `__new__` it does not need to be instantiated, but only when the class itself initialized.

I added a logger to see what would call it and here is the result:
```
2024-04-27 07:57:21,704 1489 WARNING ? odoo.addons.hw_drivers.driver: DRIVER CREATE Driver 
Stack (most recent call last):
  File "/home/pi/odoo/odoo-bin", line 8, in <module>
    odoo.cli.main()
  File "/home/pi/odoo/odoo/cli/command.py", line 66, in main
    o.run(args)
  File "/home/pi/odoo/odoo/cli/server.py", line 180, in run
    main(args)
  File "/home/pi/odoo/odoo/cli/server.py", line 173, in main
    rc = odoo.service.server.start(preload=preload, stop=stop)
  File "/home/pi/odoo/odoo/service/server.py", line 1345, in start
    load_server_wide_modules()
  File "/home/pi/odoo/odoo/service/server.py", line 1248, in load_server_wide_modules
    odoo.modules.module.load_openerp_module(m)
  File "/home/pi/odoo/odoo/modules/module.py", line 471, in load_openerp_module
    __import__('odoo.addons.' + module_name)
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1149, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/pi/odoo/addons/hw_drivers/__init__.py", line 8, in <module>
    from . import driver
  File "<frozen importlib._bootstrap>", line 1234, in _handle_fromlist
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1149, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/pi/odoo/addons/hw_drivers/driver.py", line 30, in <module>
    class Driver(Thread, IoTObjectInfo, metaclass=DriverMetaClass):
  File "/home/pi/odoo/addons/hw_drivers/driver.py", line 23, in __new__
    _logger.warning(f"DRIVER CREATE {clsname}", stack_info=True)
```